assets/src/components/map/utilities/reactDivIcon.tsx

```diff
@@ -23,26 +23,14 @@ const defaultOptions = {}
 export function useReactDivIcon(options?: DivIconOptions) {
   const opts = options || defaultOptions
   // Persistent element for react portal to use between `divIcon` recreations
-  const [iconContainer, setIconContainer] = useState<HTMLDivElement | null>(
-    null
+  const iconContainer = useMemo(() => createPortalElement(), [])
```

Comment:
Would this work?
```suggestion
  const iconContainer = useMemo(createPortalElement, [])
```

I was also wondering if we could just get rid of the `useMemo()` call and make it
```ts
const iconContainer = createPortalElement()
```

but I'm thinking that it's important that `createPortalElement()` is only called once?

Reply:
It's _very_ important that `createPortalElement()` is only called once because that's the element we hand to leaflet to manipulate, we could change it to `usePortalElement`, but it'd still be _some_ react state somewhere.

Reply:
okay I'm not actually _certain_ that we couldn't do `const iconContainer = createPortalElement()` but I personally think we should keep it as is.

Reply:
It's a good question though! Any invariants that need to be upheld by only calling this once _should_ be documented.

Reply:
I get a few test failures when I outright remove it, not sure about if those could be fixed or not but I'm not going to spend more time on it.